Bug fixes in C_Intrinsics/main.c

mouse07410 opened the pull request:
And other refinements

Commit:
Update README.md

Commit:
Added missing function definition

Commit:
Work In Progress (doesn't work yet) - attempt to get raw assembly code to compile on MacOS

Commit:
Polished Makefile and made it working on MacOS

Commit:
Changed reference to GCC

Commit:
Multiple minor fixes

Commit:
Added "-Weror" to C_Intrinsics build

Commit:
Merge branch 'master' of llcad-github.llan.ll.mit.edu:uri/AES-GCM-SIV

Commit:
Delete compiled binaries from source repo

Commit:
Add .gitignore

Commit:
Added more compiled binaries to ignore

Commit:
Minor changes

Commit:
Fixed compiler warnings, minor refactoring

Commit:
Ignore more files

Commit:
Added some VSCode stuff

Commit:
More spelling stuff

Commit:
Removed duplicate function definition

Commit:
Don't know what this was

Commit:
Add words to dictionary

Commit:
Fixed a bug, and indentation

Commit:
Added Usage comment

Commit:
Cleaned main.c source

Commit:
Moved GCM_SIV definitions to .h file

Commit:
Refactored some code out

Commit:
Addressed some strict warnings from Clang

Commit:
Eliminating compiler warnings (work in progress)

Commit:
Eliminating compiler warnings (WIP)

Commit:
Eliminating compiler warnings (WIP)

Commit:
Eliminated compiler warnings for AES256 C_Intrinsics code

Commit:
Improved output generator

Commit:
Added binaries to .gitignore

Commit:
Deleted compiled binaries from the repo

Commit:
Made "-Wa,-q" conditional - for Darwin only, and only for GCC

Commit:
Improved Makefile detection of OS and compiler

ghost commented:
You made the project work with macOS. Is there a way to make it work under iOS too?

mouse07410 commented:
> Is there a way to make it work under iOS too?

I don't know. First, I've no idea how MacOS intrinsics translate to iOS - they may or may not exist there. Second - most of my attention was on (a) refining the Reference code, and (b) polishing the C_Intrinsics. 

I was unable to do anything useful with Performance code - no clue what kind of assembler it was written for. No matter which one I tried, or with what flags - it kept failing. So, I decided to stay with the two above code versions.

Note, that this PR is somewhat outdated, and rather imperfect. I've improved the code since, fixed a bug that I introduced in the Reference `main.c`. Other than that, it should be good-to-go on MacOS and Linux.

Please let me know if you get something working on iOS, as eventually I hope to get to it.

mouse07410 commented:
OK, just forget it.
